tests/backup/backup_resiliency_test.go

```diff
@@ -1558,3 +1558,284 @@ var _ = Describe("{ScaleDownPxBackupPodWhileBackupAndRestoreIsInProgress}", func
+				schBackupName, err = GetFirstScheduleBackupName(ctx, scheduleName, orgID)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching the name of the first schedule backup - %s", schBackupName))
+				backupNames = append(backupNames, schBackupName)
+				scheduleNames = append(scheduleNames, scheduleName)
+
+				log.Infof("Taking manual backup of application with rule's")
+				backupName = fmt.Sprintf("%s-%s-%v", BackupNamePrefix, namespace, time.Now().Unix())
+				preRuleUid, _ := Inst().Backup.GetRuleUid(orgID, ctx, preRuleNameListMap[namespace])
+				postRuleUid, _ := Inst().Backup.GetRuleUid(orgID, ctx, postRuleNameListMap[namespace])
+				err = CreateBackup(backupName, SourceClusterName, bkpLocationName, backupLocationUID,
+					[]string{namespace}, labelSelectors, orgID, srcClusterUid, preRuleNameListMap[namespace], preRuleUid, postRuleNameListMap[namespace], postRuleUid, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Taking backup %s of application- %s", backupName, namespace))
+				backupNames = append(backupNames, backupName)
+
+				log.Infof("Taking schedule backup of application with rule's")
+				scheduleName = fmt.Sprintf("%s-%s-%v", BackupNamePrefix, namespace, time.Now().Unix())
+				schPolicyUid, _ = Inst().Backup.GetSchedulePolicyUid(orgID, ctx, periodicPolicyName)
+				preRuleUid, _ = Inst().Backup.GetRuleUid(orgID, ctx, preRuleNameListMap[namespace])
+				postRuleUid, _ = Inst().Backup.GetRuleUid(orgID, ctx, postRuleNameListMap[namespace])
+				err = CreateScheduleBackup(scheduleName, SourceClusterName, bkpLocationName, backupLocationUID, []string{namespace},
+					labelSelectors, orgID, preRuleNameListMap[namespace], preRuleUid, postRuleNameListMap[namespace], postRuleUid, periodicPolicyName, schPolicyUid, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verification of creating schedule backup with schedule name - %s", scheduleName))
+				schBackupName, err = GetFirstScheduleBackupName(ctx, scheduleName, orgID)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching the name of the first schedule backup - %s", schBackupName))
+				backupNames = append(backupNames, schBackupName)
+				scheduleNames = append(scheduleNames, scheduleName)
+				backupNamesMap[namespace] = backupNames
+				threadCount = threadCount + len(backupNames)
+			}
+			log.Infof("The Map of backups taken are: %v", backupNamesMap)
+		})
+
+		var sem = make(chan struct{}, threadCount)
+		var wg sync.WaitGroup
+		Step("Restoring the backed up applications", func() {
+			log.InfoD("Restoring the backed up applications")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range appNamespaces {
+				namespaceMapping := make(map[string]string)
+				for _, backupName := range backupNamesMap[namespace] {
+					sem <- struct{}{}
+					restoreName = fmt.Sprintf("%s-%s", restoreNamePrefix, backupName)
+					restoreNames = append(restoreNames, restoreName)
+					time.Sleep(1 * time.Second)
```

Comment:
why do we sleep here?

Reply:
Since this is parallel execution of restore's, sleep is added such that new namespace will have different timestamp suffixed.

Reply:
oh yeah, makes sense...